src/modules/tunnel/unabto_tunnel_common.c

```diff
@@ -49,6 +49,13 @@ void unabto_tunnel_deinit_tunnels()
 }
 
 void unabto_tunnel_stream_accept(unabto_stream* stream) {
+#if NABTO_ENABLE_TUNNEL_OPEN_ACL_CHECK
+    nabto_connect* con = unabto_stream_connection(stream);
```

Comment:
unabto_stream_close(stream); is not enough the stream also needs to be released. The best way to ensure that is to set the state to t->state = TS_CLOSING